In [1]:
import numpy as np
import pandas as pd
import sqlite3
import os

In [2]:
v13_path = '/Users/yoachim/Temp/sims_featureScheduler_runs_1.3/maf'
v13_dirs = ['baseline_2snap_v1.3_10yrs',
            'baseline_nomix_v1.3_10yrs',
            'baseline_v1.3_10yrs',
            'baseline_v1.3_1yrs',
           'wfd_95_v1.3_10yrs',
           'wfd_90_v1.3_10yrs',
           'wfd_85_v1.3_10yrs',
           'wfd_80_v1.3_10yrs',
           'wfd_75_v1.3_10yrs',
           'wfd_70_v1.3_10yrs',
           'wfd_65_v1.3_10yrs',
           'delayedrolling_mod2_sdf_0.10_v1.3_10yrs']

dirs = [os.path.join(v13_path, filename) for filename in v13_dirs]
snaps = [1]*len(v13_dirs)
snaps[0] = 2
pairs = ['mixed']*len(v13_dirs)
pairs[1] = 'same'

dirs.append('/Users/yoachim/Temp/old_runs/baseline2018a_note')
snaps.append(2)
pairs.append('same')

In [3]:
v12_path = '/Users/yoachim/gitRepos/sims_featureScheduler_runs/baselines'
v12_dirs = ['baseline_1exp_pairsame_10yrs',
           'baseline_1exp_pairsmix_10yrs',
           'baseline_2exp_pairsame_10yrs',
           'baseline_2exp_pairsmix_10yrs']

for directory in v12_dirs:
    dirs.append(os.path.join(v12_path, directory))
    
snaps.extend([1,1,2,2])
pairs.extend(['same', 'mixed', 'same', 'mixed'])

In [4]:
results = {'fONv MedianNvis':[], 'fOArea': [], 'osf':[], 'nexp':[], 'fONv MinNvis':[]}
for directory in dirs:
    conn = sqlite3.connect(os.path.join(directory, 'resultsDb_sqlite.db'))
    df = pd.read_sql('select summaryValue from summarystats where summaryName="fONv MedianNvis";', conn)
    results['fONv MedianNvis'].append(np.max(df.values))
    
    df = pd.read_sql('select summaryValue from summarystats where summaryName="fOArea";', conn)
    results['fOArea'].append(np.max(df.values))
    
    df = pd.read_sql('select summaryValue from summarystats where metricid=3;', conn)
    results['osf'].append(np.max(df.values))
    
    df = pd.read_sql('select summaryValue from summarystats where metricid=5;', conn)
    results['nexp'].append(np.max(df.values))
    
    df = pd.read_sql('select summaryValue from summarystats where summaryName="fONv MinNvis";', conn)
    results['fONv MinNvis'].append(np.max(df.values))
    
    conn.close()

In [5]:
keys = list(results.keys())
header = '| Run Name  |    '
for key in keys:
    header+= '%s |' % key

header += 'snaps |'
header += 'pairs |'
print(header)
print('| -- | -- | -- | -- | --| --| --|--|')
for i in range(len(dirs)):
    rn = os.path.basename(dirs[i])
    out = '| '
    out += rn +' |'
    for key in keys:
        out +='   %.3f  |' % results[key][i]
    out += '  %i |' % snaps[i]
    out += '  %s |' % pairs[i]
    print(out)
    

| Run Name  |    fONv MedianNvis |fOArea |osf |nexp |fONv MinNvis |snaps |pairs |
| -- | -- | -- | -- | --| --| --|--|
| baseline_2snap_v1.3_10yrs |   828.000  |   10052.220  |   0.703  |   2019329.000  |   767.000  |  2 |  mixed |
| baseline_nomix_v1.3_10yrs |   915.000  |   18127.064  |   0.776  |   2230670.000  |   847.000  |  1 |  same |
| baseline_v1.3_10yrs |   900.000  |   18090.135  |   0.759  |   2182491.000  |   834.000  |  1 |  mixed |
| baseline_v1.3_1yrs |   94.000  |   67.983  |   0.767  |   228046.000  |   69.000  |  1 |  mixed |
| wfd_95_v1.3_10yrs |   1040.000  |   18404.031  |   0.762  |   2190893.000  |   970.000  |  1 |  mixed |
| wfd_90_v1.3_10yrs |   1001.000  |   18298.280  |   0.762  |   2189357.000  |   931.000  |  1 |  mixed |
| wfd_85_v1.3_10yrs |   962.000  |   18218.547  |   0.762  |   2189452.000  |   894.000  |  1 |  mixed |
| wfd_80_v1.3_10yrs |   922.000  |   18212.672  |   0.762  |   2190844.000  |   857.000  |  1 |  mixed |
| wfd_75_v1.3_10yrs |   884

| Run Name  |    fONv MedianNvis |fOArea |osf |nexp |fONv MinNvis |snaps |pairs |
| -- | -- | -- | -- | --| --| --|--|
| baseline_2snap_v1.3_10yrs |   828.000  |   10052.220  |   0.703  |   2019329.000  |   767.000  |  2 |  mixed |
| baseline_nomix_v1.3_10yrs |   915.000  |   18127.064  |   0.776  |   2230670.000  |   847.000  |  1 |  same |
| baseline_v1.3_10yrs |   900.000  |   18090.135  |   0.759  |   2182491.000  |   834.000  |  1 |  mixed |
| baseline_v1.3_1yrs |   94.000  |   67.983  |   0.767  |   228046.000  |   69.000  |  1 |  mixed |
| wfd_95_v1.3_10yrs |   1040.000  |   18404.031  |   0.762  |   2190893.000  |   970.000  |  1 |  mixed |
| wfd_90_v1.3_10yrs |   1001.000  |   18298.280  |   0.762  |   2189357.000  |   931.000  |  1 |  mixed |
| wfd_85_v1.3_10yrs |   962.000  |   18218.547  |   0.762  |   2189452.000  |   894.000  |  1 |  mixed |
| wfd_80_v1.3_10yrs |   922.000  |   18212.672  |   0.762  |   2190844.000  |   857.000  |  1 |  mixed |
| wfd_75_v1.3_10yrs |   884.000  |   18088.457  |   0.763  |   2192487.000  |   827.000  |  1 |  mixed |
| wfd_70_v1.3_10yrs |   851.000  |   15814.810  |   0.763  |   2193207.000  |   813.000  |  1 |  mixed |
| wfd_65_v1.3_10yrs |   831.000  |   10551.600  |   0.764  |   2194390.000  |   795.000  |  1 |  mixed |
| delayedrolling_mod2_sdf_0.10_v1.3_10yrs |   880.000  |   16123.670  |   0.763  |   2192506.000  |   788.000  |  1 |  mixed |
| baseline2018a_note |   915.000  |   18056.563  |   0.716  |   2372700.000  |   836.000  |  2 |  same |
| baseline_1exp_pairsame_10yrs |   986.000  |   18226.940  |   0.785  |   2263993.000  |   914.000  |  1 |  same |
| baseline_1exp_pairsmix_10yrs |   963.000  |   18186.654  |   0.768  |   2216026.000  |   895.000  |  1 |  mixed |
| baseline_2exp_pairsame_10yrs |   907.000  |   18096.850  |   0.725  |   2092909.000  |   839.000  |  2 |  same |
| baseline_2exp_pairsmix_10yrs |   887.000  |   17942.420  |   0.710  |   2048865.000  |   821.000  |  2 |  mixed |